File: paysphere_backend/cph/src/main/java/com/test/cph/entity/Account.java
package com.test.cph.entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

import java.time.LocalDateTime;

@Entity
@Table(name = "accounts")
@Getter
@Setter
public class Account {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "account_id")
    private Long accountId;

    private String userId;

    private String accountHolderName;

    @Column(unique = true, nullable = false)
    private String accountNumber;

    @Enumerated(EnumType.STRING)
    private AccountType accountType;

    @ManyToOne
    @JoinColumn(name = "customer_id", referencedColumnName = "customerId", nullable = false)
    private Customer customer;

    private Double balance;

    private String currency;

    @Enumerated(EnumType.STRING)
    private AccountStatus status;

    @ManyToOne
    @JoinColumn(name = "branch_id", nullable = false)
    private Branch branch;

    private String ifscCode;

    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;

    @PrePersist
    public void onCreate() {
        this.createdAt = LocalDateTime.now();
        this.updatedAt = LocalDateTime.now();
        this.status = AccountStatus.ACTIVE;
    }

    @PreUpdate
    public void onUpdate() {
        this.updatedAt = LocalDateTime.now();
    }


}

File: paysphere_frontend/src/app/accounts/account.ts
import { AccountStatus } from "../models/account-status.enum";
import { AccountType } from "../models/account-type.enum";
import { Branch } from "../models/branch";
import { Customer } from "../models/customer";

export class Account {
  accountId!: number;
  userId!: number;
  accountHolderName!: string;
  accountNumber!: string;
  accountType!: AccountType;
  customer!: Customer;
  balance!: number;
  currency!: string;
  status!: AccountStatus;
  branchName!: Branch;
  createdAt!: string;   // ISO date string
  updatedAt!: string;
}